react/src/components/atoms/forms/InputNumber/index.js

```diff
@@ -80,13 +79,14 @@ const NumberInput = (props) => {
           const handleChange = (e) => {
             e.persist();
             const inputEl = ref.current;
-            const newValue = inputEl.value ? Number(inputEl.value) : inputEl.value;
-            const updateError = displayErrorMessage(newValue);
-            context.updateState({ value: newValue, ...updateError }, () => {
-              if (is.fn(props.onChange)) {
-                props.onChange(e, newValue, props.id);
-              }
-            });
+            const newValue = !is.number(Number(inputEl.value)) ? 0 : Number(inputEl.value);
```

Comment:
this will never let you set a null value -- should switch it back to the way it was i believe
```suggestion
            const newValue = inputEl.value ? Number(inputEl.value) : inputEl.value;
```